src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledArrayBuilder`1.cs

```diff
@@ -22,6 +22,7 @@ namespace Microsoft.AspNetCore.Razor.PooledObjects;
 ///  continue to be used, even if the arrays shrinks and has fewer than 4 elements.
 /// </summary>
 [NonCopyable]
+[CollectionBuilder(typeof(PooledArrayBuilder), "Create")]
```

Comment:
Do we want to change some pooled arrays to use collection expressions to demonstrate it works?

Reply:
I did locally (which is how I discovered that the non-copy analyzer needed to be updated), but I didn't want to touch existing code.